File: CS/ImageSliderWithWatermark/Code/ImageHelper.cs
using System;
using System.Collections.Generic;
using System.Drawing;
using System.IO;
using System.Linq;
using System.Web;

namespace ImageSliderWithWatermark.Code
{
    public class ImageHelper
    {
        private static Image GetImageFromUrl(string url)
        {
            var image = Image.FromFile(HttpContext.Current.Server.MapPath(url));
            return image;
        }

        private static byte[] ImageToByteArray(Image image)
        {
            using (var ms = new MemoryStream())
            {
                image.Save(ms, image.RawFormat);
                return ms.ToArray();
            }
        }

        public static byte[] GetImageWithWatermark(string url, string text, string fontName)
        {
            var image = GetImageFromUrl(url);
            PhotoUtils.DrawWatermarkText(image, text, fontName);
            return ImageToByteArray(image);
        }
    }
}

File: VB/ImageSliderWatermark/Code/ImageHelper.vb
Imports System.IO
Imports System.Drawing

Public Class ImageHelper
    Private Shared Function GetImageFromUrl(ByVal url As String) As Image
        Dim image = Drawing.Image.FromFile(HttpContext.Current.Server.MapPath(url))
        Return image
    End Function

    Private Shared Function ImageToByteArray(ByVal image As Image) As Byte()
        Using ms = New MemoryStream()
            image.Save(ms, image.RawFormat)
            Return ms.ToArray()
        End Using
    End Function

    Public Shared Function GetImageWithWatermark(ByVal url As String, ByVal text As String, ByVal fontName As String) As Byte()
        Dim image = GetImageFromUrl(url)
        PhotoUtils.DrawWatermarkText(image, text, fontName)
        Return ImageToByteArray(image)
    End Function
End Class
